fxa/oauth.py

```diff
@@ -239,12 +260,39 @@ def verify_token(self, token, scope=None):
             resp = None
 
         if resp is None:
-            url = '/verify'
-            body = {
-                'token': token
-            }
-            resp = self.apiclient.post(url, body)
-
+            # TODO we want to fetch
+            # https://oauth.accounts.firefox.com/.well-known/openid-configuration
+            # and then get the jwks_uri key to get the /jwks url, but we'll
+            # just hardcosde it like this for now until we are closer to
+            # production. We need to do this because oauth.accounts.firefox.com
+            # does not yet respond to .well-known, but accounts.firefox.com
+            # does; for consistency, oauth.accounts.firefox.com should respond
+            # to .well-known.
+
+            keys = self.apiclient.get('/jwks').get('keys', [])
+            resp = None
+            try:
+                for k in keys:
+                    try:
+                        resp = self._verify_jwt_token(json.dumps(k), token)
+                        break
+                    except jwt.exceptions.InvalidSignatureError:
+                        # It's only worth trying other keys in the event of
+                        # `InvalidSignature`; if it was invalid for other reasons
+                        # (e.g. it's expired) then using a different key won't
+                        # help. OTOH you might prefer to let the server do a bit of
+                        # extra work just to keep the control-flow simpler.
+                        continue
+            except (jwt.exceptions.DecodeError, jwt.exceptions.InvalidKeyError):
+                # It wasn't a JWT at all, or it was signed using a key type we
+                # don't support. Fall back to asking the FxA server to verify.
+                pass
+            except jwt.exceptions.PyJWTError as e:
+                # Any other JWT-related failure (e.g. expired token) can
+                # immediately surface as a trust error.
+                raise TrustError({"error": str(e)})
+            if resp is None or resp.get(key) is None:
```

Comment:
I'm confused by the `resp.get(key)` here...is `key` still bound to the cache key `"fxa.oauth.verify_token:%s:%s"` from the top of the method, or is it something else? I would expect this to be a check like `resp.get("user") is None` or some other fixed key that we expect to be in the response.

My naive read of this is that we'll always fall back to calling `/verify` because `resp.get(key)` will always be `None`, but I could very easily be missing something here.

Reply:
Yes, it's still bound to "fxa.oauth.verify_token:%s:%s". You suggested I make this change; it's extremely probable that I misunderstood what you wanted?

https://github.com/mozilla/PyFxA/pull/80#discussion_r431536563

What was `k` supposed to refer to in that comment?

Reply:
Ok, yeah, I misread which line that comment was supposed to apply to, and that's why both this confusing .get(key) is here, and also the .get(k) inside the list comprehension is not there.

Reply:
Aha, that makes sense - thanks for digging back in, and sorry to not be clearer in the initial comment.